### PR TITLE
[docs] add `eas/build` function docs

### PR DESCRIPTION
**Pull request by szdziedzic**
# Why

Document the `eas/build` function

# How

Document the `eas/build` function

# Test Plan

Preview

# TODO

I still need to add the examples in example repo before it starts to link to them correctly 😅 

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/custom-builds/schema.mdx [hunk 1]
```diff
@@ -570,6 +570,97 @@ EAS provides a set of built-in reusable functions that you can use in a workflow
...
+   - if your build uses credentials (for both `internal` and `store` [distribution](/eas/json/#distribution) builds):
+     - [`eas/checkout`](#eascheckout)
+     - [`eas/install_node_modules`](#easinstall_node_modules)
+     - [`eas/get_credentials_for_build_triggered_by_github_integration`](#easget_credentials_for_build_triggered_by_github_integration)
+     - [`eas/prebuild`](#easprebuild)
+     - [`eas/configure_eas_update`](#easconfigure_eas_update)
+     - [`eas/inject_android_credentials`](#easinject_android_credentials)
+     - [`eas/configure_android_version`](#easconfigure_android_version)
+     - [`eas/run_gradle`](#easrun_gradle)
+     - [`eas/find_and_upload_build_artifacts`](#easfind_and_upload_build_artifacts)
```
Comment (amandeepmittal) [inline docs/pages/custom-builds/schema.mdx:629]:
```suggestion
```

File: docs/pages/custom-builds/schema.mdx [hunk 2]
```diff
@@ -570,6 +570,97 @@ EAS provides a set of built-in reusable functions that you can use in a workflow
...
+  href="https://github.com/expo/eas-custom-builds-example/blob/main/.eas/build/android-build-without-credentials.yml"
+/>
+<BoxLink
+  title="android-build-with-credentials.yml"
+  description="View the steps executed behind the scenes by the `eas/build` function for an Android build with credentials in our example repository."
+  Icon={GithubIcon}
+  href="https://github.com/expo/eas-custom-builds-example/blob/main/.eas/build/android-build-with-credentials.yml"
+/>
+
+Known limitations:
```
Comment (amandeepmittal) [inline docs/pages/custom-builds/schema.mdx:658]:
```suggestion
##### Known limitations
```

Convert it to a section

File: docs/pages/custom-builds/schema.mdx [hunk 3]
```diff
@@ -570,6 +570,97 @@ EAS provides a set of built-in reusable functions that you can use in a workflow
...
+
+- This function won't configure updates for you properly if you still use the legacy classic updates. You should use the EAS Update instead. Check our [migration guide](/eas-update/migrate-from-classic-updates/) for more information.
```
Comment (amandeepmittal) [inline docs/pages/custom-builds/schema.mdx:660]:
```suggestion
- This function doesn't configure legacy classic updates. If you are using classic updates, [migrate to EAS Update](/eas-update/migrate-from-classic-updates/) before using this function.
```

File: docs/pages/custom-builds/schema.mdx [hunk 4]
```diff
@@ -570,6 +570,97 @@ EAS provides a set of built-in reusable functions that you can use in a workflow
...
+- Currently it doesn't accept any inputs and the resolved build process will be configured based on your build profile from [**eas.json**](/eas/json/) settings.
```
Comment (amandeepmittal) [inline docs/pages/custom-builds/schema.mdx:661]:
```suggestion
- It doesn't accept any inputs, and the resolved build process will be configured based on your build profile from [**eas.json**](/eas/json/).
```

File: docs/pages/custom-builds/schema.mdx [hunk 5]
```diff
@@ -570,6 +570,97 @@ EAS provides a set of built-in reusable functions that you can use in a workflow
...
+- The build process produced by `eas/build` is not configurable and you can't customize it. If you need to customize the build process you should use the subset of functions and steps that are executed behind the scenes by this function and configure them manually in your YAML configuration file as shown in the examples above.
```
Comment (amandeepmittal) [inline docs/pages/custom-builds/schema.mdx:662]:
```suggestion
- The build process produced by `eas/build` is not configurable and you can't customize it. If you need to customize the build process, use the subset of functions and steps that are executed behind the scenes by this function and configure them manually in the YAML configuration file, as shown in the examples above.
```

- Which examples is the last line referring to?

Comment (szdziedzic) [reply docs/pages/custom-builds/schema.mdx:661]:
```md
##### Android

When a build configuration is using [`withoutCredentials`](/eas/json/#withoutcredentials):

- [`eas/checkout`](#eascheckout)
- [`eas/install_node_modules`](#easinstall_node_modules)
- [`eas/prebuild`](#easprebuild)
- [`eas/configure_eas_update`](#easconfigure_eas_update)
- [`eas/run_gradle`](#easrun_gradle)
- [`eas/find_and_upload_build_artifacts`](#easfind_and_upload_build_artifacts)

When a build configuration uses credentials (for both `internal` and `store` [distribution](/eas/json/#distribution) builds):

- [`eas/checkout`](#eascheckout)
- [`eas/install_node_modules`](#easinstall_node_modules)
- [`eas/get_credentials_for_build_triggered_by_github_integration`](#easget_credentials_for_build_triggered_by_github_integration)
- [`eas/prebuild`](#easprebuild)
- [`eas/configure_eas_update`](#easconfigure_eas_update)
- [`eas/inject_android_credentials`](#easinject_android_credentials)
- [`eas/configure_android_version`](#easconfigure_android_version)
- [`eas/run_gradle`](#easrun_gradle)
- [`eas/find_and_upload_build_artifacts`](#easfind_and_upload_build_artifacts)


##### iOS

When a build configuration is using [`withoutCredentials`](/eas/json/#withoutcredentials) or [`simulator`](/eas/json/#simulator):
- [`eas/checkout`](#eascheckout)
- [`eas/install_node_modules`](#easinstall_node_modules)
- [`eas/prebuild`](#easprebuild)
- Install pods using the `pod install` command
- [`eas/configure_eas_update`](#easconfigure_eas_update)
- [`eas/generate_gymfile_from_template`](#easgenerate_gymfile_from_template)
- [`eas/run_fastlane`](#easrun_fastlane)
- [`eas/find_and_upload_build_artifacts`](#easfind_and_upload_build_artifacts)


When a build configuration uses credentials (for both `internal` and `store` [distribution](/eas/json/#distribution) builds):

- [`eas/checkout`](#eascheckout)
- [`eas/install_node_modules`](#easinstall_node_modules)
- [`eas/get_credentials_for_build_triggered_by_github_integration`](#easget_credentials_for_build_triggered_by_github_integration)
- [`eas/resolve_apple_team_id_from_credentials`](#easresolve_apple_team_id_from_credentials)
- [`eas/prebuild`](#easprebuild)
- Install pods using the `pod install` command
- [`eas/configure_eas_update`](#easconfigure_eas_update)
- [`eas/configure_ios_credentials`](#easconfigure_ios_credentials)
- [`eas/configure_ios_version`](#easconfigure_ios_version)
- [`eas/generate_gymfile_from_template`](#easgenerate_gymfile_from_template)
- [`eas/run_fastlane`](#easrun_fastlane)
- [`eas/find_and_upload_build_artifacts`](#easfind_and_upload_build_artifacts)
```

I meant these examples

Comment (amandeepmittal) [reply docs/pages/custom-builds/schema.mdx]: Thanks for the clarification!

**Comment by amandeepmittal** [comment]
> Oh I think that `####` header is smaller than standard text size 😅
> 
> Is it supposed to be like that?
> 
> cc: @Simek @amandeepmittal
> 
> Screenshot 2024-02-26 at 16 52 24

That's how H5 has been in our styles.

**Comment by amandeepmittal** [comment]
> So it's fine if I use it?

I think its fine to use it since we also use it other places. An alternative we can use here is also bold the text like **Android** or **iOS**. This won't create the section but an H5 header is also not included in the page's TOC by default (and I don't think we meant to include it originally). I think it is ideal for us to highlight them as platforms on this page so it allows quick readability.

**Review by amandeepmittal — APPROVED** [review]

**Review by sjchmiela — APPROVED** [review]

**Pull request merged**